### ISSUE TITLE
Update/rethink the allowed HTML tags in the filtered HTML text format

### ISSUE THREAD

**Issue by MrHaroldA** [backdrop/backdrop-issues#981]
Currently, a user with access to filtered HTML can create stuff like blockquotes, but can't add any headings.

Current list: `<a> <em> <strong> <cite> <blockquote> <code> <ul> <ol> <li> <dl> <dt> <dd>`

I'm not sure what this list hould be, but I guess H1-H3 should be included, and cite, blockquote and code might be removed from this list.

I ran into this when setting a empty text on the frontpage View to serve as front page after installation #144.

Thoughts?

**Comment by klonos** [comment]
I think h1 is intentionally left out because most websites use that for the page title. You don't want users abusing that tag and overriding the semantic structure of your pages. Not sure about h2/h3 but I think more or less along the same lines as h1.

**Comment by MrHaroldA** [comment]
Ok, I understand why H1 is left out, but H2/H3 (and H4) should be available to write an article, basic page, and/or "fancy" empty texts for Views.

**Comment by docwilmot** [comment]
I think some header tags should be allowed, for reasons @MrHaroldA suggest. Sites can limit the appearance of these tags in content with CSS. e.g `.content .h1 {font-size: 1.5em; // If site h1 is 2.5em for example.`

**Comment by MrHaroldA** [comment]
IIRC, you can have as many H2/3/4's on your website as you'd like (as long as the nesting is correct), but you only may have one H1.

But I'm not an HTML expert...

**Comment by MrHaroldA** [comment]
As I already expected; I was wrong ... Multiple H1's is no problem in the HTML5-era: http://webdesign.tutsplus.com/articles/the-truth-about-multiple-h1-tags-in-the-html5-era--webdesign-16824

... but then again; HTML5 also introduces a lot of other tags, like <article>, <header>, etc.

Point being: H1/H2/H3/H4 add to the document structure, and thus improves SEO. The question now is which headers should Backdrop support out of the box?

**Comment by quicksketch** [comment]
Even though it's possible to use multiple H1s (headings can "start-over" when inside of a different article or section), I don't think this is very common yet. The reason being, it's difficult to style consistently (or so I would believe). Most themes and CSS assume a certain size for h1, h2, h3, etc and generally use that size everywhere. With multiple h1s, you end can easily end up with `article h1`, `.block h1`, `h1.page-title`, and others all being styled drastically different. Maybe I'm overstating the problem, but it's still not common.

That said, I definitely agree that some headings should be allowed by default. With h1 usually the page title, h2 usually the node title (in a teaser listing), I think we should allow h3, h4, and h5 in the filtered_html text format by default. What do you think?

**Comment by docwilmot** [comment]
Yes, I agree with that.

**Comment by MrHaroldA** [comment]
Agreed!

Now we just have to wait until I figured out how to create a pull request on GitHub ;)

**Comment by quicksketch** [comment]
Yay, excellent!

> Now we just have to wait until I figured out how to create a pull request on GitHub ;)

We've got a video at https://backdropcms.org/develop/pull-requests :smile:

**Comment by MrHaroldA** [comment]
Working on it! While I'm at it; should I include `<p>` too?*

*) Rhetorical question? ;)

**Comment by MrHaroldA** [comment]
Ok, after failing to see that settings.php was also committed to my Backdrop issue branch, I guess I managed to create my first pull request! ;)

**Comment by quicksketch** [comment]
Well done! Could you [rebase your commits](https://backdropcms.org/develop/pull-requests#clean-requests) into a single commit? Tiny nitpick, we try to use sentences as our commit messages, so the capitalization would be 

```
Issue #981: Add h3, h4, h5 and p to the filtered_html allowed tags, without changing settings.php.
```

After fixing your commit, force push the branch to GitHub and the PR will update to reflect the new changes.

**Comment by quicksketch** [comment]
> While I'm at it; should I include `<p>` too?

Yes. Let's wait on `<img>` however. There are some security concerns with displaying external images.

**Comment by MrHaroldA** [comment]
Rebase done! Tnx!

**Comment by quicksketch** [comment]
Perfect! This was in my TODO list for WYSIWYG handling as well (since we need `<p>` tags to work with CKEditor's content filtering). Let's make sure the tests pass, but it looks good to me!

**Comment by klonos** [comment]
...and tests pass! Thanx for this.

**Comment by quicksketch** [comment]
I've merged this into 1.x. I don't think this is a "bug fix" and so it shouldn't go into the 1.1.x branch. It'll be in 1.2.0 that comes out Sept 15.
